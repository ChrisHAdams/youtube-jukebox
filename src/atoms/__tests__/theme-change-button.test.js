import React from "react"
import renderer from "react-test-renderer"
import {render, fireEvent, cleanup} from 'react-testing-library'
import {toHaveStyle} from 'jest-dom';
import ThemeChangeButton from "../theme-change-button";

expect.extend({toHaveStyle});

describe("ThemeChangeButton", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ThemeChangeButton/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

it('calls "onClick" prop on button click', () => {
  // Render new instance in every test to prevent leaking state
  const onClick = jest.fn();
  const { getByTestId } = render(<ThemeChangeButton onClick={onClick}/>);

  fireEvent.click(getByTestId('themeChangeButton'));
  expect(onClick).toHaveBeenCalled();

});