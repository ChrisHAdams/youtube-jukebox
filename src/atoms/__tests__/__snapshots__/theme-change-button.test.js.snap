// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeChangeButton renders correctly 1`] = `
<button
  className="sc-bdVaJa fwQqyE"
  data-testid="themeChangeButton"
>
  Change Theme
</button>
`;
